### PR TITLE
r.resamp.filter: implement parallelization with OpenMP

### PR DESCRIPTION
**Pull request by aaronsms**
Checklists before merging:
- [x] code review
- [x] CI passes
- [x] performance section in documentation
- [x] confirm values in test are from the old version (run new tests with old code)
- [x] run tests without OpenMP
- [x] visual check of results with custom data ("looks good" with non NC SPM dataset)
- [x] check that it works with really large data (16B cells)
- [x] run multi-core benchmark (no degraded performance with many threads)
- [x] run one core benchmark on many resolutions or many cell
- [x] rebase to main

### CODE REVIEW

**Review by marisn — APPROVED** [review]
Didn't test the performance, but the code and generated output looks good.

File: raster/r.resamp.filter/main.c [hunk 1]
```diff
@@ -152,16 +155,18 @@ struct filter {
...
 static struct Cell_head dst_w, src_w;
 static double f_x_radius, f_y_radius;
 static int row_scale, col_scale;
-static DCELL *inbuf;
+static DCELL **inbuf;
 static DCELL *outbuf;
-static DCELL **bufs;
+static DCELL ***bufs;
+static int bufrows;
+static int nprocs;
```
Comment (HuidaeCho) [inline raster/r.resamp.filter/main.c:169]: Delete this line if not used.
```
main.c:169:12: warning: ‘nprocs’ defined but not used [-Wunused-variable]
 static int nprocs;
```

File: raster/r.resamp.filter/main.c [hunk 2]
```diff
@@ -378,15 +407,16 @@ int main(int argc, char *argv[])
...
+        *radius, *x_radius, *y_radius, *memory, *nprocs;
     } parm;
     struct
     {
 	struct Flag *nulls;
     } flag;
     char title[64];
-    int i;
+    int i, t;
+    int nprocs;
```
Comment (HuidaeCho) [inline raster/r.resamp.filter/main.c:419]: Or did you mean to use the static `nprocs` above?

**Comment by petrasovaa** [comment]
Thank you for the review, I will merge now. #2692 needs to be again updated.

**Comment by nilason** [comment]
> Thank you for the review, I will merge now. #2692 needs to be again updated.

Done.

**Pull request merged**